pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -387,6 +393,25 @@ func WriteToFile(config clientcmdapi.Config, filename string) error {
 	return nil
 }
 
+func lockFile(filename string) error {
+	// TODO: find a way to do this with actual file locks. Will
+	// probably need seperate solution for windows and linux.
+	f, err := os.OpenFile(lockName(filename), os.O_CREATE|os.O_EXCL, 0)
```

Comment:
Ah thanks. Looks like other folks are using windows API directly - https://github.com/pinterest/knox/blob/master/client/flock_windows.go

@krousey Thanks for the explanation